modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -156,70 +162,70 @@ public synchronized List<ServiceRegistration<?>> deploy(
 							"enabled", "false"
 						).put(
 							"osgi.jaxrs.application.select",
-							"(osgi.jaxrs.name=" + objectDefinition.getName() +
-								")"
+							"(osgi.jaxrs.name=" + osgiJAXRsName + ")"
 						).put(
 							"osgi.jaxrs.extension", "true"
 						).put(
 							"osgi.jaxrs.name",
 							objectDefinition.getName() +
-								"ObjectDefinitionContextProvider"
+								objectDefinition.getCompanyId() +
+									"ObjectDefinitionContextProvider"
 						).build()),
 					_bundleContext.registerService(
 						ExceptionMapper.class,
 						new ObjectEntryManagerHttpExceptionMapper(),
 						HashMapDictionaryBuilder.<String, Object>put(
 							"osgi.jaxrs.application.select",
-							"(osgi.jaxrs.name=" + objectDefinition.getName() +
-								")"
+							"(osgi.jaxrs.name=" + osgiJAXRsName + ")"
 						).put(
 							"osgi.jaxrs.extension", "true"
 						).put(
 							"osgi.jaxrs.name",
 							objectDefinition.getName() +
-								"ObjectEntryManagerHttpExceptionMapper"
+								objectDefinition.getCompanyId() +
+									"ObjectEntryManagerHttpExceptionMapper"
 						).build()),
 					_bundleContext.registerService(
 						ExceptionMapper.class,
 						new ObjectEntryValuesExceptionMapper(),
 						HashMapDictionaryBuilder.<String, Object>put(
 							"osgi.jaxrs.application.select",
-							"(osgi.jaxrs.name=" + objectDefinition.getName() +
-								")"
+							"(osgi.jaxrs.name=" + osgiJAXRsName + ")"
 						).put(
 							"osgi.jaxrs.extension", "true"
 						).put(
 							"osgi.jaxrs.name",
 							objectDefinition.getName() +
-								"ObjectEntryValuesExceptionMapper"
+								objectDefinition.getCompanyId() +
+									"ObjectEntryValuesExceptionMapper"
 						).build()),
 					_bundleContext.registerService(
 						ExceptionMapper.class,
 						new ObjectValidationRuleEngineExceptionMapper(),
 						HashMapDictionaryBuilder.<String, Object>put(
 							"osgi.jaxrs.application.select",
-							"(osgi.jaxrs.name=" + objectDefinition.getName() +
-								")"
+							"(osgi.jaxrs.name=" + osgiJAXRsName + ")"
 						).put(
 							"osgi.jaxrs.extension", "true"
 						).put(
 							"osgi.jaxrs.name",
 							objectDefinition.getName() +
-								"ObjectValidationRuleEngineExceptionMapper"
+								objectDefinition.getCompanyId() +
```

Comment:
All this concatenations... should we add a separator in between? something like `ObjectValidationRuleEngineExceptionMapper#122323`.

Reply:
Ok, so, to have a common pattern. I would change the IDs to have something like
* `1234#University`
* `1234#ObjectValidationRuleEngineExceptionMapper`
Being `1234` the company ID.

Why at the beginning and not at the end?
Because concatenate before a # information is a pattern that we are already using to provide information to [batch engine.](https://github.com/liferay/liferay-portal/blob/410123945a2bd55e8d58aa8e0baf51cd2235e4d0/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java#L256)

I think is better but maybe is just my perception.

Reply:
Forget about the previous comment. I would use your pattern. My pattern does not follow the Bundle-SymbolicName syntax

Reply:
I'm afraid that, for now, we should forget about the `#` as seems like the `Bundle-SymbolicName` doesn't allow strange characters. 
I'm following your pattern  but without `#`